packages/finance/src/tests/test_vesting_linear.cairo

```diff
@@ -0,0 +1,201 @@
+use openzeppelin_access::ownable::interface::{IOwnableDispatcher, IOwnableDispatcherTrait};
+use openzeppelin_finance::tests::common::{VestingStrategy, TestData, setup};
+use openzeppelin_finance::tests::mocks::vesting_mocks::LinearVestingMock;
+use openzeppelin_finance::vesting::VestingComponent::InternalImpl;
+use openzeppelin_finance::vesting::VestingComponent;
+use openzeppelin_finance::vesting::interface::{IVestingDispatcher, IVestingDispatcherTrait};
+use openzeppelin_test_common::vesting::VestingSpyHelpers;
+use openzeppelin_testing::constants::OWNER;
+use openzeppelin_testing::events::EventSpyExt;
+use snforge_std::{spy_events, start_cheat_caller_address, start_cheat_block_timestamp_global};
+use starknet::ContractAddress;
+
+//
+// Setup
+//
+
+type ComponentState = VestingComponent::ComponentState<LinearVestingMock::ContractState>;
+
+fn COMPONENT_STATE() -> ComponentState {
+    VestingComponent::component_state_for_testing()
+}
+
+fn TEST_DATA() -> TestData {
+    TestData {
+        strategy: VestingStrategy::Linear,
+        total_allocation: 200,
+        beneficiary: OWNER(),
+        start: 30,
+        duration: 100,
+        cliff_duration: 0
+    }
+}
+
+//
+// initializer
+//
```

Comment:
Small nit: we're not really going through fn by fn. I commented on this already (and think it might be a good idea to provide some more unit-y tests); however, if we don't include them, we can probably remove this since we're not following the convention for the rest of the test mod

Reply:
Same applies for the preset test mod, either way

Reply:
Changed it to "Tests" to separate from the "Setup" block